crates/iceberg/src/transaction/mod.rs

```diff
@@ -129,8 +124,10 @@ impl Transaction {
             }
         };
         let mut snapshot_id = generate_random_id();
+
+        // todo revisit this, this check won't be 100% valid because it's not checking the staging table
         while self
-            .current_table
+            .table
             .metadata()
             .snapshots()
             .any(|s| s.snapshot_id() == snapshot_id)
```

Comment:
Maybe we should move the generation  of `snapshot_id` to `SnapshotProducer::new`. The catch of doing this: if retry happens, every attemp will have a different `snapshot_id`. I have not seen any side effects yet